src/core.js

```diff
@@ -332,19 +331,59 @@ const Editable = module.exports = class Editable {
     }
   }
 
-  // Highlight text within an editable.
-  //
-  // The first occurrence of the provided 'text' will be highlighted.
-  //
-  // The markup used for the highlighting will be removed from
-  // the final content.
-  //
-  // @param editableHost {DomNode}
-  // @param text {String}
-  // @param highlightId {String} Optional
-  //   Added to the highlight markups in the property `data-word-id`
-  highlight ({editableHost, text, highlightId}) {
-    return highlightSupport.highlightText(editableHost, text, highlightId)
+  /**
+   * Highlight text within an editable.
+   *
+   * By default highlights all occurences of `text`.
+   * Pass it a `textRange` object to highlight a
+   * specific text portion.
+   *
+   * The markup used for the highlighting will be removed
+   * from the final content.
+   *
+   *
+   * @param  {Object} options
+   * @param  {DOMNode} options.editableHost
+   * @param  {String} options.text
+   * @param  {String} options.highlightId Added to the highlight markups in the property `data-word-id`
+   * @param  {Object} [options.textRange] An optional range which gets used to set the markers.
+   * @param  {Number} [options.textRange.start]
```

Comment:
I think `start` and `end` are not optional if a `textRange` object is passed, right? So I think we can remove the square brackets around the param name here and in the next line.

Reply:
Done!